Spec/Auth.swift

```diff
@@ -2620,104 +2620,84 @@ class Auth : QuickSpec {
-                // init ARTRest
-                let restOptions = AblyTests.setupOptions(AblyTests.jsonRestOptions)
-                let rest = ARTRest(options: restOptions)
 
-                // get first token
-                let tokenParams = ARTTokenParams()
-                tokenParams.capability = "{\"wrongchannel\": [\"*\"]}"
-                tokenParams.clientId = "testClientId"
+        describe("Reauth") {
 
-                var firstToken = ""
+            // RTC8
+            it("should use authorise({force: true}) to reauth with a token with a different set of capabilities") {
+                let options = AblyTests.commonAppSetup()
+                let initialToken = getTestToken(capability: "{\"foo\":[\"*\"]}")
+                options.token = initialToken
+                let realtime = ARTRealtime(options: options)
+                defer { realtime.dispose(); realtime.close() }
 
                 waitUntil(timeout: testTimeout) { done in
-                    rest.auth.requestToken(tokenParams, withOptions: nil) { tokenDetails, error in
-                        expect(error).to(beNil())
-                        expect(tokenDetails).toNot(beNil())
-                        firstToken = tokenDetails!.token
+                    realtime.connection.on(.Connected) { stateChange in
+                        expect(stateChange?.reason).to(beNil())
                         done()
                     }
                 }
-                expect(firstToken).toNot(beNil())
-                expect(firstToken.characters.count > 0).to(beTrue())
 
-                // init ARTRealtime
-                let realtimeOptions = AblyTests.commonAppSetup()
-                realtimeOptions.token = firstToken
-                realtimeOptions.clientId = "testClientId"
+                let authOptions = ARTAuthOptions()
+                authOptions.force = true
 
-                let realtime = ARTRealtime(options:realtimeOptions)
-                defer { realtime.dispose(); realtime.close() }
+                let tokenParams = ARTTokenParams()
+                tokenParams.capability = "{\"restricted\":[\"*\"]}"
 
-                // wait for connected state
                 waitUntil(timeout: testTimeout) { done in
-                    realtime.connection.once(.Connected) { stateChange in
-                        expect(stateChange!.reason).to(beNil())
-                        expect(stateChange?.current).to(equal(realtime.connection.state))
+                    realtime.auth.authorise(tokenParams, options: authOptions) { tokenDetails, error in
+                        expect(error).to(beNil())
+                        expect(tokenDetails).toNot(beNil())
                         done()
                     }
-                    realtime.connect()
                 }
 
-                // create a `rightchannel` channel and check can't attach to it
-                let channel = realtime.channels.get("rightchannel")
```

Comment:
Why did you remove this?  This is the point of the test no, i.e. attach with one set of permissions, change them, check that permissions have actually changed by checking the channel attach success?